juriscraper/opinions/united_states_backscrapers/state/sd.py

```diff
@@ -1,229 +1,60 @@
-                    case_names.append("O'Bryan v. Ashland")
-                elif "j.d.m.c." in s.lower() and self.year == 2007:
-                    case_names.append("In Re the Matter of J.D.M.C.")
-                elif "argus" in s.lower() and self.year == 2007:
-                    case_names.append("Argus Leader v. Hagen")
-                elif "nedved" in s.lower() and self.year == 2007:
-                    case_names.append(
-                        "Countryside S. Homeowners Assoc. v. Nedved"
-                    )
-                elif "gisi" in s.lower() and self.year == 2007:
-                    case_names.append("Gisi v. Gisi")
-                elif "western" in s.lower() and self.year == 2007:
-                    case_names.append("Great Western Bank v. H&E Enterprises")
-                elif "canyon" in s.lower() and self.year == 2007:
-                    case_names.append("Jackson v. Canyon Place Homeowners")
-                elif "vansteenwyk" in s.lower() and self.year == 2007:
-                    case_names.append("Vanteenwyk v. Baumgartner Trees")
-                elif "jewett" in s.lower() and self.year == 2011:
-                    case_names.append("Jewett v. Real Tuff")
-                elif "jensen" in s.lower() and self.year == 2011:
-                    case_names.append("State v. Jensen")
-                elif "cook" in s.lower() and self.year == 2011:
-                    case_names.append("Orr v. Cook")
-                elif "living" in s.lower() and self.year == 2011:
-                    case_names.append("Benson Living Trust")
-                else:
-                    raise AttributeError
-        return case_names
-
-    def _get_case_dates(self):
-        path = '//tr[contains(@id, "ctl00xmainCopyxWGOpinions_r")]/td[1]/@uv'
-        return [
-            datetime.strptime(date_string, "%m/%d/%Y").date()
-            for date_string in self.html.xpath(path)
-        ]
-
-    def _get_precedential_statuses(self):
-        return ["Published"] * len(self.case_names)
-
-    def _get_docket_numbers(self):
-        path = '//tr[contains(@id, "ctl00xmainCopyxWGOpinions_r")]//a[2]/@href'
-        docket_numbers = []
-        for url in self.html.xpath(path):
+        self.status = "Published"
+        self.url = "https://ujs.sd.gov/Supreme_Court/opinions.aspx"
+
+    def _set_parameters(self):
+        self.method = "POST"
+        self.parameters = {
+            "ctl00$ctl00$ScriptManager1": "ctl00$ctl00$ContentPlaceHolder1$ChildContent1$UpdatePanel_Opinions|ctl00$ctl00$ContentPlaceHolder1$ChildContent1$LinkButton_Next",
+            "__EVENTTARGET": "ctl00$ctl00$ContentPlaceHolder1$ChildContent1$LinkButton_Next",
+        }
+        self.parameters["__VIEWSTATE"] = (
+            self.html.xpath("//*[@id='__VIEWSTATE']/@value")[0],
+        )
+        self.parameters["__EVENTVALIDATION"] = (
+            self.html.xpath("//*[@id='__EVENTVALIDATION']/@value")[0],
+        )
```

Comment:
I missed this before, but let's get the xpath stuff out of the set parameters method, please.

Reply:
√